lesson_5/lesson_5.py

```diff
@@ -0,0 +1,45 @@
+# coded: UTF-8
+
+# 1 Task
+
+import re
+
+input_value = input('Зробіть введення значення: ')
+
+string_regex = "^[a-zA-Zа-яА-ЯёЁЄєІіЇї]+$"
+symbol_regex = "^[&^_@#\-!$%^&*()£€¥§°¢_+|~=`{}\[\]:\";'<>?,.\/\\\]+$"
+number_regex = "^\d+$"
+
+string_pattern = re.compile(string_regex)
+symbol_pattern = re.compile(symbol_regex)
+number_pattern = re.compile(number_regex)
+
+if len(input_value) == 1:
+    if number_pattern.search(input_value):
+        print('Це “число”')
+        if int(input_value) % 2 == 0:
+            print(f'Число “{input_value}“ парне.')
+        else:
+            print(f'Число “{input_value}“ НЕ парне.')
+
+    elif string_pattern.search(input_value):
```

Comment:
there is a better way to check if a character or a string is non a digit: `not str.isalnum()`. Why did you decide to use regular expressions here?

Reply:
Yes of course I can do someting like this:

input = input()

for symbol in input:
    if symbol.isdigit():
        print(f"{symbol} is number")
    elif not symbol.isalnum():
        print(f"{symbol} is symbol")
    elif symbol.isalpha():
        print(f"{symbol} is letter")
    else:
        print(f"What you insert?")

But - this is not interesting)

At start I tryed do Match/Case with function - that doesn't work, so I think "maybe i can do this with regular expressions", after that i feld sad because it still not worked and in the end i use RegExs because i allready done it. Of course i can use function isdigit() - but when i already maked Regular Expression for Symbol and Text - i wanted make one-piece structure, and think this is good idea and practice for me. (This is the Main reason Why I use regular expressions)